crates/driver/src/domain/competition/solution/scoring.rs

```diff
@@ -86,15 +89,26 @@ impl Trade {
     ///
     /// Denominated in SURPLUS token
     fn surplus(&self) -> Option<eth::Asset> {
+        let (sell_amount, buy_amount) = match self.policies.first() {
+            Some(order::FeePolicy::PriceImprovement {
+                factor: _,
+                max_volume_factor: _,
+                quote,
+            }) => adjust_quote_to_order_limits(
+                self.sell.amount.0,
+                self.buy.amount.0,
+                self.side,
+                quote.sell.amount.0,
+                quote.buy.amount.0,
+                quote.fee.amount.0,
+            )
+            .unwrap(),
+            _ => (self.sell.amount.0, self.buy.amount.0),
+        };
```

Comment:
I assume, for the price improvement policy, adjusted sell/buy amounts should be used instead of the full ones.

Reply:
`surplus` function is also used for native_surplus calculation which is supposed to always use order limit amounts. So this would require a bit of refactoring similar to what we already have with surplus_over_reference_price